src/getMigrateDashboards.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  DashboardState as DashboardState50,
+  ContentRecord,
+} from "@activeviam/activeui-sdk-5.0";
+import { DashboardState as DashboardState51 } from "@activeviam/activeui-sdk-5.1";
+import _omit from "lodash/omit";
+import {
+  ErrorReport,
+  MigrateFromVersions,
+  OutcomeCounters,
+} from "./migration.types";
+import { _addErrorToReport } from "./_addErrorToReport";
+import { _getMapOfFolderIds } from "./_getMapOfFolderIds";
+import { _serializeError } from "./_serializeError";
+
+type MigrateDashboardCallback<fromVersion extends MigrateFromVersions> = (
+  dashboardState: fromVersion extends "5.0" ? DashboardState50 : any,
+) => fromVersion extends "5.0" ? DashboardState51 : any;
+
+/**
+ * Handles the generic logic to migrate dashboards, specifically the errors.
+ * `cb` is a generic callback that is meant to handle dashboards migration from one specific version to another.
+ * Mutates `migratedUIFolder`, `dashboards`, `counters` and `errorReport`.
+ */
+export const getMigrateDashboards = <fromVersion extends MigrateFromVersions>(
+  {
+    legacyUIFolder,
+    migratedUIFolder,
+    errorReport,
+    counters,
+    doesReportIncludeStacks,
+  }: {
+    legacyUIFolder: ContentRecord;
+    migratedUIFolder: ContentRecord;
+    errorReport: ErrorReport;
+    counters: OutcomeCounters;
+    doesReportIncludeStacks: boolean;
+  },
+  cb: MigrateDashboardCallback<fromVersion>,
+): void => {
+  let migratedDashboard;
+
+  const legacyContent =
+    legacyUIFolder.children?.dashboards.children?.content.children;
+  const legacyStructure =
+    legacyUIFolder?.children?.bookmarks?.children?.structure!;
```

Comment:
I don't understand the changes here.
- If this is intended to work for 4.3, this definition of `legacyContent` should not work: there is no `ui/dashboards` folder in 4.3.
- If this is intended to work for 5.0, then this definition of `legacyStructure` shold not work: there is no `ui/bookmarks` in 5.0.
- It doesn't make sense to mix both 4.3 and 5.0.

We discussed that 4.3 to 5.0 is a special case that cannot really be genericized anyway, and so that:
- `getMigrateDashboards` would assume the ActiveUI 5 content server structure, and be used to migrate for 5.x to 5.y
- 4.3 to 5.0 would handle this logic on its own, by mutating `contentServer` directly.

Am I missing something ?

Reply:
Yes, you're completely right, that's my bad! I fixed that